### PR TITLE
[CSS] Additional information for style containment relating to counters and quotes

### PR DESCRIPTION
**Pull request by bsmth**
This PR adds some details for `contain: style;` which applies style containment for `counters` and `quotes` in CSS.

Additional pages have been touched to refer back to the contain page where it would prove helpful.

#### Related issues

- [x] Relates to https://github.com/mdn/interactive-examples/pull/2259
- [x] __Depends on__ https://github.com/mdn/css-examples/pull/87

### CODE REVIEW

File: files/en-us/web/css/contain/index.md [hunk 1]
```diff
@@ -191,6 +191,72 @@ This also means that the first image no longer floats down to the second article
...
+.contain {
+  contain: style;
+}
+```
+
+{{EmbedGHLiveSample("css-examples/contain/contain-style-counter.html", '100%', 500)}}
+
+Without containment, the counter would increment from 1 to 4 for each heading.
+Style containment causes the `counter-increment` to be scoped to the element's subtree and the counter begins again at 1.
+CSS quotes are affected similarly in that the [`content`](/en-US/docs/Web/CSS/content) values relating to quotes are scoped to the element:
```
Comment (estelle) [inline files/en-us/web/css/contain/index.md:228]:
```suggestion
CSS quotes are similarly affected in that the [`content`](/en-US/docs/Web/CSS/content) values relating to quotes are scoped to the element:
```

File: files/en-us/web/css/contain/index.md [hunk 2]
```diff
@@ -191,6 +191,72 @@ This also means that the first image no longer floats down to the second article
 
 {{EmbedGHLiveSample("css-examples/contain/contain-fix.html", '100%', 500)}}
 
+### Using the style value for containment
+
+Style containment scopes [counters](/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters) and [quotes](/en-US/docs/Web/CSS/quotes) to the contained element.
+For CSS counters, the `counter-increment` and `counter-set` properties are scoped to the element as if it's at the root of the document.
```
Comment (estelle) [inline files/en-us/web/css/contain/index.md:197]:
```suggestion
For CSS counters, the `counter-increment` and `counter-set` properties are scoped to the element as if the element is at the root of the document.
```

File: files/en-us/web/css/contain/index.md [hunk 3]
```diff
@@ -122,11 +122,11 @@ article {
...
 {{EmbedGHLiveSample("css-examples/contain/simple-layout.html", '100%', 400)}}
 
-You can immediately see an issue — no effort is made to clear the floating beyond the bottom of the article.
+There is an issue because the floating element is laid out beyond the bottom of the article.
 
-### Float interference
+### Adding an additional floating element
 
-If we were to insert another image at the bottom of the first article, a large portion of the DOM tree may be re-laid out or repainted, and this would interfere further with the layout of the second article:
+To the previous example, if we insert another image at the bottom of the first article, a large portion of the DOM tree is re-laid out or repainted, and this interferes with the layout of the second article:
```
Comment (estelle) [inline files/en-us/web/css/contain/index.md:129]:
```suggestion
In the previous example, if we insert another image at the bottom of the first article, a large portion of the DOM tree is re-laid out or repainted, and this interferes with the layout of the second article:
```

**Review by estelle — APPROVED** [review]
Thank you!

**Comment by bsmth** [comment]
Hi @estelle thanks for your assistance, there's one snag; this PR depends on another one that's open in the `css-examples` repo. The content we've merged here references examples that don't exist yet, unfortunately. The dependent PR will be reviewed and merged shortly, but it's something to look out for in future.

Thanks 🙌🏻

**Pull request merged**